### DISCUSSION TITLE
Playground - allow for dragging and dropping of shapes etc.

### DISCUSSION THREAD

**Discussion by benkoshy** [terrastruct/d2#597] [Feature Requests]
Hi Folks

I'll throw in my two cents.

The domain specific knowledge required to play around in the sandbox or playground is extreme - I would have to know the scripting language itself.

IT would be nice if the playground had a UI which did not require any scipting at all - I want to be able to drag and drop shapes and arrows onto a canvas, much like photoshop or MS paint, and have the script be generated automatically. perhaps this is too difficult, but it would be very handy to have a crude version of this available. I wouldn't spend too much time on this, it was just a 'shower thought'.

![image](https://user-images.githubusercontent.com/15097447/210288942-f1dba6e0-5d77-4c5a-9a0c-d7d0ed86745c.png)

**Comment by alixander** [comment]
@benkoshy indeed, this is how it works in the Terrastruct app. it is a looot of extra logic though, and not something easy to implement in the immediate future on playground.

**Reply by benkoshy** [reply]
Very good. Today I learned.

(Out of curiosity - what is the Terrastruct app written in?)

**Reply by alixander** [reply]
same stuff. Go and JS
